### PR TITLE
Add functions: isDecimalOverflow(), countDigits()

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `countDigits(x)` function that count number of decimal digits in integer or decimal column. Add `isDecimalOverflow(d, [p])` function that checks if the value in Decimal column is out of its (or specified) precision.

Detailed description / Documentation draft:
ClickHouse checks interger overflow in operations with Decimal types. So there could be implicit decimal overflows (without integer ones). Then we could not extract value to other DBMS or even serialize to strings and back without decimal overflow. This function makes it possible to find such values.
`isDecimalOverflow(dec)` with one argument checks values in `Decimal(P, S)` column and rerutn 1 if count of digits in underlying integer is greater than P. When present with 2 arguments `isDecimalOverflow(dec, p)` uses second argument as precision to check overflow. Form with 2 arguments could be helpful for extraction to another DBMS or file.

### CODE REVIEW

**Comment by 4ertus2** [comment]
Related #14124

File: src/Functions/countDigits.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+/// Returns 1 if and Decimal value has more digits then it's Precision allow, 0 otherwise.
```
Comment (alexey-milovidov) [inline src/Functions/countDigits.cpp:20]: Wrong comment.

File: src/Functions/countDigits.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+public:
+    static constexpr auto name = "countDigits";
+
+    static FunctionPtr create(const Context &)
+    {
+        return std::make_shared<FunctionCountDigits>();
+    }
+
+    String getName() const override { return name; }
+    bool useDefaultImplementationForNulls() const override { return false; }
```
Comment (alexey-milovidov) [inline src/Functions/countDigits.cpp:33]: Why not?

File: src/Functions/countDigits.cpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        auto result_column = ColumnUInt8::create();
+
+        auto call = [&](const auto & types) -> bool
+        {
+            using Types = std::decay_t<decltype(types)>;
+            using Type = typename Types::RightType;
+            using ColVecType = std::conditional_t<IsDecimalNumber<Type>, ColumnDecimal<Type>, ColumnVector<Type>>;
+
+            if (const ColumnConst * const_column = checkAndGetColumnConst<ColVecType>(src_column.column.get()))
```
Comment (alexey-milovidov) [inline src/Functions/countDigits.cpp:61]: useDefaultImplementationForConstants?

**Review by alexey-milovidov — COMMENTED** [review]
It should be stated (somewhere) that `digits` counts also trailing zeros after decimal point.

**Comment by 4ertus2** [comment]
> It should be stated (somewhere) that `digits` counts also trailing zeros after decimal point.

Done in new comment to FunctionCountDigits class.

**Comment by alexey-milovidov** [comment]
Unbundled broken in master.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**